internal/db/embedding.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/philippgille/chromem-go"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/errors"
+	"github.com/sourcenetwork/defradb/internal/keys"
+)
+
+func getEmbeddingFunc(provider, model, url string) (chromem.EmbeddingFunc, error) {
+	switch provider {
+	case "ollama":
+		return chromem.NewEmbeddingFuncOllama(model, url), nil
+	case "openai":
+		normalized := true
+		apiURL := url
+		if apiURL == "" {
+			apiURL = chromem.BaseURLOpenAI
+		}
+		return chromem.NewEmbeddingFuncOpenAICompat(
+			apiURL,
+			os.Getenv("OPENAI_API_KEY"),
+			model,
+			&normalized,
+		), nil
+	default:
+		return nil, NewErrUnknownEmbeddingProvider(provider)
+	}
+}
+
+// setEmbedding sets the embedding fields on the document if the related fields are dirty.
+func (c *collection) setEmbedding(ctx context.Context, doc *client.Document, isCreate bool) error {
+	embeddingGenerated := false
+	for _, colField := range c.Definition().GetFields() {
+		if colField.Embedding != nil {
```

Comment:
suggestion: Inverting this `if` and `continue`-ing would remove quite a long indentation, and IMO make the code a bit easier to read.

Reply:
related to my suggestion above regarding moving the `EmbeddingDescription` to the `CollectionDescription` instead of the `CollectionFieldDescription` would simplify this function, where it can go straight to the embeddings rather than scanning all the fields to see if there is a description on them.

Reply:
Inverted the `if` and moved the embedding description to the collection description.